File: backend/src/main/java/com/myfavsdb/dto/MovieSearchResult.java
package com.myfavsdb.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public class MovieSearchResult {
    
    @JsonProperty("id")
    public Long id;
    
    @JsonProperty("title")
    public String title;
    
    @JsonProperty("name")
    public String name; // Para séries
    
    @JsonProperty("original_title")
    public String originalTitle;
    
    @JsonProperty("original_name")
    public String originalName; // Para séries
    
    @JsonProperty("overview")
    public String overview;
    
    @JsonProperty("poster_path")
    public String posterPath;
    
    @JsonProperty("media_type")
    public String mediaType;
    
    @JsonProperty("vote_average")
    public Double voteAverage;
    
    @JsonProperty("release_date")
    public String releaseDate;
    
    @JsonProperty("first_air_date")
    public String firstAirDate; // Para séries
    
    public MovieSearchResult() {}
    
    public String getDisplayTitle() {
        return title != null ? title : name;
    }
    
    public String getDisplayOriginalTitle() {
        return originalTitle != null ? originalTitle : originalName;
    }
    
    public String getDisplayDate() {
        return releaseDate != null ? releaseDate : firstAirDate;
    }
    
    public String getFullPosterUrl() {
        if (posterPath != null && !posterPath.isEmpty()) {
            return "https://image.tmdb.org/t/p/w500" + posterPath;
        }
        return null;
    }
    
    public String getMediaTypeDisplay() {
        if ("movie".equals(mediaType)) {
            return "Filme";
        } else if ("tv".equals(mediaType)) {
            return "Série";
        }
        return mediaType;
    }
} 

File: MyFavsDB/lib/src/models/movie_search_result.dart
class MovieSearchResult {
  final int id;
  final String? title;
  final String? name;
  final String? originalTitle;
  final String? originalName;
  final String? overview;
  final String? posterPath;
  final String? mediaType;
  final double? voteAverage;
  final String? releaseDate;
  final String? firstAirDate;

  MovieSearchResult({
    required this.id,
    this.title,
    this.name,
    this.originalTitle,
    this.originalName,
    this.overview,
    this.posterPath,
    this.mediaType,
    this.voteAverage,
    this.releaseDate,
    this.firstAirDate,
  });

  factory MovieSearchResult.fromMap(Map<String, dynamic> json) {
    return MovieSearchResult(
      id: json['id'] ?? 0,
      title: json['title'],
      name: json['name'],
      originalTitle: json['original_title'],
      originalName: json['original_name'],
      overview: json['overview'],
      posterPath: json['poster_path'],
      mediaType: json['media_type'],
      voteAverage: json['vote_average']?.toDouble(),
      releaseDate: json['release_date'],
      firstAirDate: json['first_air_date'],
    );
  }

  String get displayTitle {
    return title ?? name ?? 'Sem título';
  }

  String get displayOriginalTitle {
    return originalTitle ?? originalName ?? '';
  }

  String get displayDate {
    return releaseDate ?? firstAirDate ?? '';
  }

  String? get fullPosterUrl {
    if (posterPath != null && posterPath!.isNotEmpty) {
      return 'https://image.tmdb.org/t/p/w500$posterPath';
    }
    return null;
  }

  String get mediaTypeDisplay {
    switch (mediaType) {
      case 'movie':
        return 'Filme';
      case 'tv':
        return 'Série';
      default:
        return mediaType ?? 'Desconhecido';
    }
  }
} 